lib/services/northBound/northboundServer.js

```diff
@@ -117,7 +117,9 @@ function start(config, callback) {
 function stop(callback) {
     logger.info(context, 'Stopping IoT Agent');
 
-    northboundServer.server.close(callback);
+    if (northboundServer) {
+        northboundServer.server.close(callback);
+    }
```

Comment:
In case it's stopped, the callback must be called, anyways, or the IOTA will get stuck.